### DISCUSSION TITLE
Use fetch for dynamic route generation

### DISCUSSION THREAD

**Discussion by simeon9696** [nuxt/nuxt#9459] [Questions (Nuxt 2)]
Apologies for this question but I'm a bit lost how on to do a full static site generation with dynamic routes using fetch().

Usually, in `nuxt.config.js` I'd pass an array of routes and payloads via `generate.routes` and access the payload in the page component in `asyncData`. But how would I do the the same using the fetch hook? Should the fetch hook be used in this case? 

Just for some context it's an e-commerce site and I just have to generate pages for the different products and I'm using firebase.

**Comment by kissu** [comment]
You can totally use the `fetch()` hook, it is actually more flexible than `asyncData()` in a lot of ways.

This article may help you: https://fr.nuxtjs.org/blog/understanding-how-fetch-works-in-nuxt-2-12/

And this page of the documentation: https://nuxtjs.org/docs/2.x/features/data-fetching/

**Reply by simeon9696** [reply]
Thanks for the links!

How can I access the payload from `generate.routes` in the `fetch()` hook though? I ask because I saw this warning in the docs you mentioned: 
![image](https://user-images.githubusercontent.com/31748055/123498484-27bc6680-d5fe-11eb-9b72-ecc04ac9daba.png)

And I don't want to build my app with features that are going to be deprecated soon i.e. passing in a context argument for the fetch hook. 

Or am I misunderstanding this entirely?
